### ISSUE TITLE
Please consider renaming the integer types for clarity

### ISSUE THREAD

**Issue by chris-morgan** [AssemblyScript/assemblyscript#104]
The integer types have two names: the canonical C-style name (`sbyte`, `byte`, `short`, `ushort`, `int`, `uint`, `long`, `ulong`) and an alias that actually specifies what it *is* (`int8`, `uint16`, `int64`, _&c._).

I’m not likely to actually ever _use_ AssemblyScript, but I urge you all the same: please kill the C-style names; they’re simply confusing, with a legacy of befuddlement and uncertainty. Make the names that indicate the size the only names, then people will be happier.

---

Going one step further, I suggest changing `int8`, `uint16` _et al._ to `i8`, `u16` _et al._ This matches the naming scheme used by Rust (`i8`, `u16`, `i64`, `f32`, _&c._), and the native WebAssembly types (`i32`, `i64`, `f32` and `f64`), though potentially confusingly so seeing as the WebAssembly integer types are not inherently signed or unsigned.

(Concerning `intptr` and `uintptr`: these are C-style names that are subtly misleading, because there are a few different numbers involved—for starters, on various architectures pointer widths and word sizes are not the same; I think what you have here is not strictly “pointer size” but rather “word size”. If you would know more gnarly details, investigate the history behind the renaming of Rust’s `int` and `uint` types to `isize` and `usize`! In the end, I think the naming of these types isn’t precise or perfect, but it’s not terrible like `short` _et al._.)

**Comment by dcodeIO** [comment]
I agree from a clarity perspective. Personally, I just don't like switching between letters and numbers constantly from an ergonomic point of view (possibly simply a lack of typing skills on my side), though this introduces a valuable search-for-the-numbers micro time frame to reconsider the perfect type. So, that's when I figured that, if I abandon my C type prejudices, and dictate that an int is always 4 bytes etc., it's not thaaat bad anymore to stick to byte, short, int and long. Not sure if that's even an educated argument, but well, you know, decisions.

**Comment by nidin** [comment]
Perfect :)
